dlib/global_optimization/find_max_global_abstract.h

```diff
@@ -81,7 +92,8 @@ namespace dlib
         const max_function_calls num,
         const std::chrono::nanoseconds max_runtime = FOREVER,
         double solver_epsilon = 0,
-        const std::vector<std::vector<function_evaluation>>& initial_function_evals = {}
+        const std::vector<std::vector<function_evaluation>>& initial_function_evals = {},
+        stop_condition should_stop
```

Comment:
This needs some kind of default, like `[](double){ return false; }`

Reply:
Actually, `stop_condition should_stop = never_stop_early` is probably better.  And renaming `default_stop_condition` to `never_stop_early` and also noting it in the docs (i.e in the abstract.h file).  The word default doesn't communicate to someone what it does, but using a name like `never_stop_early` tells the reader something important about what it does.